noir_stdlib/src/eddsa.nr

```diff
@@ -11,6 +13,28 @@ pub fn eddsa_poseidon_verify(
     signature_r8_y: Field,
     message: Field
 ) -> bool {
+    let mut hasher = PoseidonHasher::default();
+    eddsa_verify_with_hasher(
+        pub_key_x,
+        pub_key_y,
+        signature_s,
+        signature_r8_x,
+        signature_r8_y,
+        message,
+        &mut hasher
+    )
+}
+
+pub fn eddsa_verify_with_hasher<H>(
+    pub_key_x: Field,
+    pub_key_y: Field,
+    signature_s: Field,
+    signature_r8_x: Field,
+    signature_r8_y: Field,
+    message: Field,
+    hasher: &mut H
+) -> bool 
+where H: Hasher {
```

Comment:
```suggestion
pub fn eddsa_verify_with_hasher<H>(
    pub_key_x: Field,
    pub_key_y: Field,
    signature_s: Field,
    signature_r8_x: Field,
    signature_r8_y: Field,
    message: Field,
) -> bool 
where H: Hasher + Default {
```

This currently allows us to inject a hasher with some preseeded state which will muck with the verification algorithm. I think that it would be best to instantiate it inside of the function.

Reply:
I agree but every time I try something like this I am getting compilation errors. It seems that Noir needs to get the hasher as an argument in order to recover the implementation.

Reply:
Ah yeah, seems like we'd need to implement turbofish in order for us to be able to do this.

Reply:
I resolve the conversation because I created an issue for it: #4514